### PR TITLE
feat: Add `title` prop for table components

### LINKED ISSUE #308: Add `title?: string | null` prop to table components with top-level title

### PR DESCRIPTION
**Pull request by xplato**
Closes #308 • Adds a `title` prop to table components with a top-level title.

### CODE REVIEW

**Comment by xplato** [comment]
@razor-x is this approach what you had in mind?

File: src/lib/seam/components/DeviceTable/DeviceTable.tsx [hunk 1]
```diff
@@ -109,9 +111,13 @@ export function DeviceTable({
...
       <TableHeader>
-        <TableTitle>
-          {t.devices} <Caption>({filteredDevices.length})</Caption>
-        </TableTitle>
+        {title != null ? (
+          <TableTitle>
+            {title ?? t.devices} <Caption>({filteredDevices.length})</Caption>
+          </TableTitle>
+        ) : (
+          <div />
```
Comment (xplato) [inline src/lib/seam/components/DeviceTable/DeviceTable.tsx:119]: Not sure if we want this. Preserves the location of the search field.

Comment (razor-x) [reply src/lib/seam/components/DeviceTable/DeviceTable.tsx:119]: Ah, yea good catch on the layout quirk. I don't love an empty div as a solution though, should be via class name if possible 🤔

Comment (xplato) [reply src/lib/seam/components/DeviceTable/DeviceTable.tsx]: Using `<div className='seam-fragment' />` for now. If you want a different className and/or extracted to a atomic component, let me know!

File: src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx [hunk 2]
```diff
@@ -140,9 +142,13 @@ export function AccessCodeTable({
       <ContentHeader onBack={onBack} />
       <TableHeader>
         <div className='seam-left'>
-          <TableTitle>
-            {t.accessCodes} <Caption>({filteredAccessCodes.length})</Caption>{' '}
-          </TableTitle>
+          {title != null ? (
+            <TableTitle>
+              {t.accessCodes} <Caption>({filteredAccessCodes.length})</Caption>{' '}
```
Comment (xplato) [inline src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx:147]: Do we want to add another prop (in a separate PR) for hiding the device count as well?

Comment (razor-x) [reply src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx:147]: Interesting so if they override the title then there is no way to see device count. I think for this first version it's ok.

We could either allow the title prop to be a template literal that allows them to specify a placeholder for the count, or we could add a new prop onDevices

Comment (xplato) [reply src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx:147]: > Interesting so if they override the title then there is no way to see device count. I think for this first version it's ok.

The opposite, actually. They can always see the device count unless `title === null`. I was just thinking ahead that some people _might_ wanna remove the count. We can wait until a customer asks for this though

Comment (razor-x) [reply src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx:147]: Exposing data related hooks as props probably needs more careful thought. I don't think we should add it adhoc here.

File: src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx [hunk 3]
```diff
@@ -140,9 +142,14 @@ export function AccessCodeTable({
       <ContentHeader onBack={onBack} />
       <TableHeader>
         <div className='seam-left'>
-          <TableTitle>
-            {t.accessCodes} <Caption>({filteredAccessCodes.length})</Caption>{' '}
-          </TableTitle>
+          {title != null ? (
+            <TableTitle>
+              {title ?? t.accessCodes}{' '}
+              <Caption>({filteredAccessCodes.length})</Caption>{' '}
```
Comment (razor-x) [inline src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx:148]: This nbsp is new not in the other one. Is it intentional?
```suggestion
              <Caption>({filteredAccessCodes.length})</Caption>
```

**Review by razor-x — APPROVED** [review]
Let's update the docs after merge

File: src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx (hunk 2 above)
Comment (xplato) [reply src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx]: Sounds good to me! We can push this to an issue 👍

**Pull request merged**